hypertrace-graphql-gateway-service-metric-utils/src/main/java/org/hypertrace/graphql/utils/metrics/gateway/BaselineMetricIntervalListContainer.java

```diff
@@ -0,0 +1,158 @@
+        .flatMapStream(this::flattenSeriesEntry)
+        .flatMapSingle(entry -> this.buildMergeablePartial(entry.getValue(), entry.getKey()));
+  }
+
+  private Stream<Map.Entry<MetricAggregationRequest, BaselineInterval>> flattenSeriesEntry(
+      Map.Entry<MetricSeriesRequest, BaselineMetricSeries> entry) {
+    return entry.getValue().getBaselineValueList().stream()
+        .map(interval -> Map.entry(entry.getKey().aggregationRequest(), interval));
+  }
+
+  private Single<BaselineMetricIntervalListContainer.MergeableIntervalPartial>
+      buildMergeablePartial(
+          BaselineInterval interval, MetricAggregationRequest aggregationRequest) {
+    return this.baselineAggregationConverter
+        .convert(interval.getBaseline())
+        .map(
+            aggregation ->
+                new BaselineMetricIntervalListContainer.MergeableIntervalPartial(
+                    this.extractIntervalTimeRange(interval), aggregationRequest, aggregation));
+  }
+
+  private BaselineMetricIntervalListContainer.IntervalTimeRange extractIntervalTimeRange(
+      BaselineInterval interval) {
+    return new BaselineMetricIntervalListContainer.IntervalTimeRange(
+        Instant.ofEpochMilli(interval.getStartTimeMillis()),
+        Instant.ofEpochMilli(interval.getEndTimeMillis()),
+        getMetricBaselineAggregation(interval.getBaseline()));
+  }
+
+  private MetricBaselineAggregation getMetricBaselineAggregation(Baseline baseline) {
+    if (baseline == null) {
+      return new MetricBaselineAggregationConverter.MetricBaselineAggregationDefaultInstance();
+    }
+    return new MetricBaselineAggregationConverter.MetricBaselineAggregationImpl(baseline);
+  }
+
+  @Value
+  private static class IntervalTimeRange {
+    Instant startTime;
+    Instant endTime;
+    MetricBaselineAggregation metricBaselineAggregation;
+  }
+
+  @Value
+  private static class MergeableIntervalPartial {
+    BaselineMetricIntervalListContainer.IntervalTimeRange intervalTimeRange;
+    MetricAggregationRequest aggregationRequest;
+    MetricBaselineAggregation aggregation;
+  }
+
+  @Getter
+  @Accessors(fluent = true)
+  private static class ConvertedMetricInterval extends BaselineConvertedAggregationContainer
+      implements BaselineMetricInterval {
+    final Instant startTime;
+    final Instant endTime;
+
+    ConvertedMetricInterval(
+        Map<MetricLookupMapKey, MetricBaselineAggregation> metricAggregationMap,
+        IntervalTimeRange intervalTimeRange) {
+      super(getBaselineAggMap(metricAggregationMap));
+      this.startTime = intervalTimeRange.getStartTime();
+      this.endTime = intervalTimeRange.getEndTime();
+    }
+
+    private static Map<MetricLookupMapKey, BaselinedMetricAggregation> getBaselineAggMap(
+        Map<MetricLookupMapKey, MetricBaselineAggregation> metricAggregationMap) {
+      Map<MetricLookupMapKey, BaselinedMetricAggregation> baselineMap = new HashMap<>();
+      metricAggregationMap.forEach(
```

Comment:
Processing should happen in the converter, not the data container - but this highlights the issue I discussed on the interfaces - we've added an extra layer of nesting here, where we'd have:
```
  baselineSeries: [
     {
        sum: {
           value: null // This is meaningless, it's harded coded to null below
            baseline: { // This is a redundant layer of nesting
               value: <value>
            }
        }
     }
  ]

Reply:
Fixed this and removed all processing here.